### PR TITLE
Add CPU support for pipe.py benchmarks

### PR DESCRIPTION
**Pull request by yuanyuan0057**
#  What does this PR do?
Add CPU support for benchmarks/pipe.py

### CODE REVIEW

**Comment by blefaudeux** [comment]
LGTM, @froody would you mind having a look ?

File: benchmarks/pipe.py [hunk 1]
```diff
@@ -283,7 +283,7 @@ def train(lm_dataloader, model, criterion, optimizer, vocab_size, args):
 
     num_params = reduce(operator.add, (reduce(operator.mul, x.size()) for x in model.parameters()))
     if model.group:
-        total = torch.Tensor([num_params]).cuda()
+        total = torch.Tensor([num_params]).cuda() if torch.cuda.is_available() else torch.Tensor([num_params])
```
Comment (froody) [inline benchmarks/pipe.py:286]: total = torch.Tensor([num_params])
if torch.cuda.is_available():
    total = total.cuda()

File: benchmarks/pipe.py [hunk 2]
```diff
@@ -347,15 +347,19 @@ def __len__(self):
         optimizer.zero_grad()
         try:
             if (pipe_group is None or pipe_group.rank() == 0) and not args.ddp_zero:
-                tmp = batch["input"].to(get_first_device(model))
+                tmp = batch["input"]
+                if torch.cuda.is_available():
+                    tmp = tmp.to(get_first_device(model))
```
Comment (froody) [inline benchmarks/pipe.py:352]: Can you instead modify get_first_device to return "cpu"?

**Review by froody — APPROVED** [review]
Minor nuts, otherwise LGTM, thanks!

File: benchmarks/pipe.py (hunk 2 above)
Comment (yuanyuan0057) [reply benchmarks/pipe.py]: Updated

**Comment by yuanyuan0057** [comment]
Can you please merge the commit if it looks good to you @froody ?

**Comment by blefaudeux** [comment]
> Can you please merge the commit if it looks good to you @froody ?

Done, thanks a lot @yuanyuan0057 !

**Pull request merged**